### ISSUE TITLE
only-arrow-functions should have an option to let pass IIFEs

### ISSUE THREAD

**Issue by danielrentz** [palantir/tslint#4062]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 2.9.2
- __Running TSLint via__: grunt-tslint

#### TypeScript code being linted

```ts
(async function () {
  // await something in top-level code
}());
```

with `tslint.yaml` configuration:

```yaml
extends:
- tslint:recommended
```

#### Actual behavior
Error: non-arrow functions are forbidden

#### Expected behavior
Should be allowed. Converting this naively to an arrow function:
```ts
(async () => {
  // await something in top-level code
}());
```
gives a syntax error. The function needs to have another extra pair of parentheses:
```ts
((async () => {
  // await something in top-level code
})());
```
which looks a little crowded. Note that the outer parentheses cannot be omitted without getting other errors.

**Comment by johnwiseheart** [comment]
I think the extra bracket isn't too bad, and this feels like a very specific option to make. Feel free to disable the rule if need be.
